app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,19 @@
+class InvitationsController < ApplicationController
+  before_action :require_user, only: [:new, :create]
+  def new
+    @invitation = Invitation.new
+  end
+
+  def create
+    @invitation = Invitation.new(params.require(:invitation).permit(:friend_name, :email, :message))
+    @invitation.user = current_user
+    if @invitation.save
+      AppMailer.invitation(@invitation).deliver
+      flash[:notice] = 'Your invitation was sent.'
+      redirect_to new_invitation_path
+    else
+      flash[:error] = 'Your invitation was not sent. Please check input fields.'
```

Comment:
You should use `flash.now` when you `render` otherwise the flash will pop on next get request.